standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/IMetaStoreClient.java

```diff
@@ -3227,6 +3227,15 @@ void commitTxn(CommitTxnRequest rqst)
    */
   void abortTxns(List<Long> txnids) throws TException;
 
+  /**
+   * Abort a list of transactions with additional information of
+   * errorcodes as defined in TxnErrorMsg.java.
+   * @param txnids List of transaction ids
+   * @param errorCode Error code specifying the reason for abort.
+   * @throws TException
+   */
+  void abortTxns(List<Long> txnids, long errorCode) throws TException;
```

Comment:
we should make this method generic and use Request object here as well

Reply:
Implemented. Done.